Date: Fri, 04 Oct 1996 16:38:00 +0100
From: David Coe <david@coent.demon.co.uk>
Subject: Re: SUMMARY: problems using gcc for compiling groff

This is a multi-part message in MIME format.

--------------5CC131704942
Content-Type: text/plain; charset=us-ascii
Content-Transfer-Encoding: 7bit

Thanks Dieter & David for all the spade-work with cygwin-b16 and groff.
I think the news about using the libstdc++ from B14 went down a black
hole during my ISP's mail outage. It was a really vital piece of
information for me!!! 

Further comments on Groff (and other GNU stuff) as follows:

1. The need to hack configure (reminiscent of MKS Toolkit) must be
peculiar to Win95; I've not needed to do this on NT (3.51 with B14 and
4.0 with B16) at all for any of the GNU tool-chain I've looked at.

2. Having Autoconf and Texinfo installed helps avoid meddling with the
makefiles. The latest versions compile, install and even run (:-)
out-of-the-box.

3. The Grodvi module in Groff writes binary to stdout and an appropriate
setmode is needed in dvi.cc (patch file attached).

4. The "double" cast has been broken since cygwin b13.5. A kludge in the
same patch for dvi.cc get around it for the time being. As I understand
there are other broken casts, there are MAJOR PROBLEMS for anyone doing
numerical analysis with cygwin!!!!!

5. System() is called in pic/pic.cc and troff/input.cc and is broken in
B16. It hasn't caused me problems (so far) but putting in Dave Murrell's
patch (that I borrowed for man-14h) might be a good idea.

6. If you're using Groff with Man, don't forget to modify lib/man.config
to point to groff rather than troff ... | grotty ... ! There is also no
need to pre-format the man-pages - Man automatically formats them and
tucks them away nicely compressed.

Best wishes

-- 
Dr David Coe			     \=\
58 Fairlawn Drive, East Grinstead     \=\   Tel +44 1342 326860
West Sussex, RH19 1NT, United Kingdom  \=\  Fax +44 1342 316019

--------------5CC131704942
Content-Type: text/plain; charset=us-ascii; name="patch"
Content-Transfer-Encoding: 7bit
Content-Disposition: inline; filename="patch"

*** grodvi/dvi.cc.orig	Thu Jun 22 07:42:00 1995
--- grodvi/dvi.cc	Fri Oct 04 13:34:11 1996
***************
*** 20,25 ****
--- 20,29 ----
  
  #include "driver.h"
  
+ #ifdef _WIN32
+ #include <fcntl.h>
+ #endif
+ 
  #define DEFAULT_LINEWIDTH 40
  static int linewidth = DEFAULT_LINEWIDTH;
  
***************
*** 358,364 ****
--- 362,372 ----
    dvi_font *f = output_font_table[i].f;
    out4(f->checksum);
    out4(output_font_table[i].point_size*RES_7227);
+ #ifdef __CYGWIN32__
+   out4(int((float(f->design_size)/(1<<20))*RES_7227*100 + .5));
+ #else
    out4(int((double(f->design_size)/(1<<20))*RES_7227*100 + .5));
+ #endif
    const char *nm = f->get_internal_name();
    out1(0);
    out_string(nm);
***************
*** 850,855 ****
--- 858,868 ----
    static char stderr_buf[BUFSIZ];
    setbuf(stderr, stderr_buf);
    int c;
+ 
+ #ifdef _WIN32
+   setmode( fileno( stdout ), O_BINARY );
+ #endif
+ 
    while ((c = getopt(argc, argv, "F:vw:d")) != EOF)
      switch(c) {
      case 'v':

--------------5CC131704942--

-
For help on using this list, send a message to
"gnu-win32-request@cygnus.com" with one line of text: "help".

